python_modules/libraries/dagster-components/dagster_components/core/component_decl_builder.py

```diff
@@ -115,16 +121,13 @@ def from_path(path: Path) -> "YamlComponentDecl":
 
     def get_component_type(self, registry: ComponentTypeRegistry) -> type[Component]:
         parsed_defs = self.component_file_model
-        if parsed_defs.type.startswith("."):
-            component_registry_key = parsed_defs.type[1:]
-
-            for component_type in find_local_component_types(self.path):
-                if get_component_type_name(component_type) == component_registry_key:
+        if parsed_defs.type.endswith(".py"):
+            name, py_file = parsed_defs.type.split("@")
```

Comment:
We should only have one call to `.split("@")` in the code base and pass around a data structure as much as possible